src/gtc/cuir/cuir_codegen.py

```diff
@@ -46,7 +67,15 @@ class CUIRCodegen(codegen.TemplatedGenerator):
         """
     )
 
-    FieldAccess = as_mako("${name}(${offset}${''.join(f', {i}_c' for i in _this_node.data_index)})")
+    def visit_FieldAccess(self, node: cuir.FieldAccess, **kwargs: Any):
+        kwargs["sep"] = "" if not node.data_index else ", "
+        parsed = [self.visit(index, **kwargs) for index in node.data_index]
+        kwargs["this_data_index"] = [
+            f"{index}_c" if check_int(index) else index for index in parsed
+        ]
+        return self.generic_visit(node, **kwargs)
+
+    FieldAccess = as_mako("${name}(${offset}${sep}${', '.join(this_data_index)})")
```

Comment:
```suggestion
        def maybe_const(s):
            try:
                return f"{int(s)}_c"
            except ValueError:
                return s

        this_data_index = "".join(", " + maybe_const(self.visit(index, **kwargs)) for index in node.data_index)
        return self.generic_visit(node, this_data_index=this_data_index, **kwargs)

    FieldAccess = as_mako("${name}(${offset}${this_data_index})")
```
I would go with something a bit more specific than `check_int` and put the function just here instead of the top of the file. Seems more readable to me, but not too important.